benchmark/framework/utils.go

```diff
@@ -18,35 +18,45 @@ package framework
 
 import (
 	"context"
-	"os"
+	"io"
 
 	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/reference"
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/containerd/remotes/docker"
 	"github.com/containerd/containerd/remotes/docker/config"
+	dockercliconfig "github.com/docker/cli/cli/config"
 )
 
-func (proc *ContainerdProcess) PullImageFromECR(
+func (proc *ContainerdProcess) PullImageFromRegistry(
 	ctx context.Context,
 	imageRef string,
-	platform string,
-	awsSecretFile string) (containerd.Image, error) {
+	platform string) (containerd.Image, error) {
 	opts := GetRemoteOpts(ctx, platform)
-	opts = append(opts, containerd.WithResolver(GetECRResolver(ctx, awsSecretFile)))
+	opts = append(opts, containerd.WithResolver(GetResolver(ctx, imageRef)))
 	image, pullErr := proc.Client.Pull(ctx, imageRef, opts...)
 	if pullErr != nil {
 		return nil, pullErr
 	}
 	return image, nil
 }
 
-func GetECRResolver(ctx context.Context, awsSecretFile string) remotes.Resolver {
-	username := "AWS"
-	secretByteArray, err := os.ReadFile(awsSecretFile)
-	secret := string(secretByteArray)
+func GetResolver(ctx context.Context, imageRef string) remotes.Resolver {
+
+	var username string
+	var secret string
+	refspec, err := reference.Parse(imageRef)
 	if err != nil {
-		panic("Cannot read aws ecr login password")
+		panic("Failed to parse image ref")
+	}
+	cf := dockercliconfig.LoadDefaultConfigFile(io.Discard)
+	if cf.ContainsAuth() {
+		if ac, err := cf.GetAuthConfig(refspec.Hostname()); err == nil {
+			username = ac.Username
+			secret = ac.Password
+		}
```

Comment:
Can we fall back to AWS credentials if the registry host is in AWS? Possibly in a way that can be extended later with additional credential sources?

Reply:
I'm not sure I understand.. credentials are fetched based on the registry host provided by the imageRef, the user is required to have valid credentials stored in the docker config before attempting to run the benchmark tests.

Reply:
I think it's ok to defer to the docker configuration for credentials. To convert AWS credentials (e.g. an instance profile) into ECR credentials, you would configure something like [amazon-ecr-credential-helper](https://github.com/awslabs/amazon-ecr-credential-helper)